server/service/integration_core_test.go

```diff
@@ -64,9 +64,16 @@ func (s *slowReader) Read(p []byte) (n int, err error) {
 
 func (s *integrationTestSuite) TestSlowOsqueryHost() {
 	t := s.T()
-	s.server.Config.ReadTimeout = 2 * time.Second
+	_, server := RunServerForTestsWithDS(
+		t,
+		s.ds,
+		&TestServerOpts{
+			SkipCreateTestUsers: true,
+			HTTPServerConfig:    &http.Server{ReadTimeout: 2 * time.Second},
+		},
+	)
 	defer func() {
-		s.server.Config.ReadTimeout = 25 * time.Second
+		server.Close()
 	}()
 
 	req, err := http.NewRequest("POST", s.server.URL+"/api/v1/osquery/distributed/write", &slowReader{})
```

Comment:
You are creating a new `server` up there, but then here you still use `s.server`. Is that intentional?

Reply:
not at all, thanks for the catch. Weird that the test still passed